docker-java-core/src/main/java/com/github/dockerjava/core/DefaultInvocationBuilder.java

```diff
@@ -276,11 +281,19 @@ public void put(InputStream body, MediaType mediaType) {
                 callback.onComplete();
             } catch (Exception e) {
                 callback.onError(e);
+            } finally {
+                onStart.countDown();
             }
         }, "docker-java-stream-" + Objects.hashCode(request));
         thread.setDaemon(true);
 
         thread.start();
+
+        try {
+            onStart.await();
```

Comment:
callback's `onStart` should be used to detect started operation. Blocking here is an unexpected behaviour change

Reply:
Sorry for the possible confusion: `onStart` is `CountDownLatch` here actually. I will rename it.

Reply:
my point is that `DefaultInvocationBuilder` should not be changed. Instead, the callee should use `awaitStarted()` or similar means

Reply:
Thank you, I got it now. I thought there will be inconsistency with other `post` and `get` methods of `DefaultInvocationBuilder` which do not accept `ResultCallback` as a parameter. The requests in these cases are executed in the same thread and `awaitStarted()` is not required. But this is the expected behavior?

Reply:
Those that have no `ResultCallback` as a parameter are blocking methods, while `ResultCallback`-based ones are kinda asynchronous, hence no blocking behaviour